src/zenml/integrations/skypilot/orchestrators/skypilot_base_vm_orchestrator.py

```diff
@@ -271,29 +272,36 @@ def prepare_or_run_pipeline(
                 f"sudo docker login --username $DOCKER_USERNAME --password "
                 f"$DOCKER_PASSWORD {stack.container_registry.config.uri}"
             )
-            task_envs = {
-                "DOCKER_USERNAME": docker_username,
-                "DOCKER_PASSWORD": docker_password,
-            }
+            task_envs["DOCKER_USERNAME"] = (docker_username,)
+            task_envs["DOCKER_PASSWORD"] = (docker_password,)
         else:
             setup = None
-            task_envs = None
 
         # Run the entire pipeline
 
         # Set the service connector AWS profile ENV variable
         self.prepare_environment_variable(set=True)
 
         try:
+            if isinstance(self.cloud, sky.clouds.Kubernetes):
+                image = image
+                run_command = f"/opt/venv/bin/{entrypoint_str} {arguments_str}"
```

Comment:
this seems like a weird entrypoint.. have you tested it?

Reply:
Now this is a tricky one, Not sure what would be the exact solution long term but if we give skypilot the `entrypoint`  it would use the default Python which is Miniconada but we want to use the one we have already set when building docker image in pipeline run and that python version is in `/opt/venv/bin/`

Reply:
then i would at lest make it configurable from the outside in case something weird changes

Reply:
Not just something weird, but anyone using a custom image which does not magically use the same venv.
What is skypilot doing exactly? Because just calling `python` should just use the venv/system python that's active, or do they do some magic to prevent that?